File: frontend/next-generation-ticketing-system/src/Entity/CustomEntity/ScheduleCard.ts
import BusSchedule from "../BusSchedule";
import Seat from "../Seat";
import RouteDistrict from "../RouteDistrict";

class ScheduleCard {
  busSchedule: BusSchedule;
  seats: Seat[];
  routeDistricts: RouteDistrict[];

  constructor(
    busSchedule: BusSchedule,
    seats: Seat[],
    routeDistricts: RouteDistrict[]
  ) {
    this.busSchedule = busSchedule;
    this.seats = seats;
    this.routeDistricts = routeDistricts;
  }
}

export default ScheduleCard;

File: backend/ngts/src/main/java/com/three/ngts/Entity/CustomEntity/ScheduleCard.java
package com.three.ngts.Entity.CustomEntity;

import com.three.ngts.Entity.BusSchedule;
import com.three.ngts.Entity.RouteDistrict;
import com.three.ngts.Entity.Seat;
import lombok.Data;

import java.util.List;

@Data
public class ScheduleCard {
    private BusSchedule busSchedule;
    private List<Seat> seats;
    private List<RouteDistrict> routeDistricts;

}
